src/s3path.jl

```diff
@@ -68,25 +73,27 @@ function S3Path(
     key::AbstractPath;
     isdirectory::Bool=false,
     config::AbstractAWSConfig=global_aws_config(),
+    version::S3PathVersion=nothing,
 )
     return S3Path(
         key.segments,
         "/",
         normalize_bucket_name(bucket),
         isdirectory,
         config,
+        version,
     )
 end
 
 # To avoid a breaking change.
-function S3Path(str::AbstractString; config::AbstractAWSConfig=global_aws_config())
-    result = tryparse(S3Path, str; config=config)
+function S3Path(str::AbstractString; config::AbstractAWSConfig=global_aws_config(), version=nothing)
+    result = tryparse(S3Path, str; config=config, version=version)
```

Comment:
Based on the comment above the function we should probably just not update this as it was unofficially deprecated in https://github.com/JuliaCloud/AWSS3.jl/pull/88.
```suggestion
function S3Path(str::AbstractString; config::AbstractAWSConfig=global_aws_config())
    result = tryparse(S3Path, str; config=config)
```

Reply:
does mean the non-"deprecated" way is to use the `S3Path(bucket, key)` constructor? Because I've actually never seen that constructor used before, only this `S3Path("s3://bucket/key")` version. So I'd be against the soft-deprecation-- feels disruptive for no reason.

Reply:
ah, I see this is probably so that we don't need to pass it to `tryparse` to instead parse it out of a string? I guess that makes sense

Reply:
I believe `parse(S3Path, str)` is probably the preferred way to do this. @rofinn can comment further